cadquery/cq.py

```diff
@@ -2501,11 +2516,32 @@ def cutThruAll(self, positive=False, clean=True):
 
         see :py:meth:`cutBlind` to cut material to a limited depth
         """
-        maxDim = self.largestDimension()*1.1 #for numerical stability
-        if not positive:
-            maxDim *= (-1.0)
+        wires = self.ctx.pendingWires
+        self.ctx.pendingWires = []
+
+        solidRef = self.findSolid()
+        faceRef = self.findFace()
+
+        #if no faces on the stack take the nearest face parallel to the plane zDir
+        if not faceRef:
+            #first select all with faces with good orietation
+            sel = selectors.PerpendicularDirSelector(self.plane.zDir)
+            faces = sel.filter(solidRef.Faces())
+            #then select the closest
+            sel = selectors.NearestToPointSelector(self.plane.origin.toTuple())
+            faceRef = sel.filter(faces)[0]
+
+        rv = []
+        for solid in solidRef.Solids():
+            s = solid.dprism(faceRef, wires, thruAll=True, additive=False,
+                             taper=-taper)
```

Comment:
I'm assuming the - sign here is intentional and is because of the way the OCC API works, but I want to confirm.

Reply:
Indeed, minus sign is intended.